pallets/circuit/src/machine/mod.rs

```diff
@@ -0,0 +1,516 @@
+                    XExecSignalId<T>,
+                    Vec<
+                        Vec<
+                            FullSideEffect<
+                                T::AccountId,
+                                T::BlockNumber,
+                                EscrowedBalanceOf<T, T::Escrowed>,
+                            >,
+                        >,
+                    >,
+                >(local_ctx.xtx_id, local_ctx.full_side_effects.clone());
+
+                for (_step_cnt, side_effect_id, _step_side_effect_id) in steps_side_effects_ids {
+                    <pallet::Pallet<T> as Store>::SFX2XTXLinksMap::insert::<
+                        SideEffectId<T>,
+                        XExecSignalId<T>,
+                    >(side_effect_id, local_ctx.xtx_id);
+                }
+
+                <pallet::Pallet<T> as Store>::LocalXtxStates::insert::<XExecSignalId<T>, LocalState>(
+                    local_ctx.xtx_id,
+                    local_ctx.local_state.clone(),
+                );
+                <pallet::Pallet<T> as Store>::PendingXtxTimeoutsMap::insert::<
+                    XExecSignalId<T>,
+                    T::BlockNumber,
+                >(local_ctx.xtx_id, local_ctx.xtx.timeouts_at);
+                <pallet::Pallet<T> as Store>::PendingXtxBidsTimeoutsMap::insert::<
+                    XExecSignalId<T>,
+                    T::BlockNumber,
+                >(
+                    local_ctx.xtx_id,
+                    T::SFXBiddingPeriod::get() + frame_system::Pallet::<T>::block_number(),
+                );
+                <pallet::Pallet<T> as Store>::XExecSignals::insert::<
+                    XExecSignalId<T>,
+                    XExecSignal<T::AccountId, T::BlockNumber>,
+                >(local_ctx.xtx_id, local_ctx.xtx.clone());
+
+                true
+            },
+            (CircuitStatus::PendingBidding, CircuitStatus::InBidding) => {
+                <pallet::Pallet<T> as Store>::XExecSignals::mutate(local_ctx.xtx_id, |x| {
+                    *x = Some(local_ctx.xtx.clone())
+                });
+
+                true
+            },
+            (CircuitStatus::InBidding, CircuitStatus::Ready) => {
+                <pallet::Pallet<T> as Store>::FullSideEffects::mutate(local_ctx.xtx_id, |x| {
+                    *x = Some(local_ctx.full_side_effects.clone())
+                });
+                <pallet::Pallet<T> as Store>::XExecSignals::mutate(local_ctx.xtx_id, |x| {
+                    *x = Some(local_ctx.xtx.clone())
+                });
+                // Always clean temporary PendingSFXBids and TimeoutsMap after bidding
+                <pallet::Pallet<T> as Store>::PendingSFXBids::remove_prefix(local_ctx.xtx_id, None);
+                <pallet::Pallet<T> as Store>::PendingXtxBidsTimeoutsMap::remove(local_ctx.xtx_id);
+
+                true
+            },
+            (_, CircuitStatus::Killed(_cause)) => {
```

Comment:
How do you feel about defining these without wildcards? So this one should be `PendingBidding | InBidding` or something like that?